pyhive/presto.py

```diff
@@ -316,6 +317,8 @@ def _process_response(self, response):
             self._state = self._STATE_FINISHED
         if 'error' in response_json:
             raise DatabaseError(response_json['error'])
+        if 'id' in response_json:
```

Comment:
could you please add a unit test for this change?

Reply:
Sure - how would you like that done given that you're not really testing the json form of the responses in the current tests (at least as far as I could see)?

Reply:
2 ways I can see it can be tackled:
1. integration test, e.g. trigger the this condition via presto query and check if cursor has this id, maybe somewhere here: https://github.com/dropbox/PyHive/blob/65076bbc8697a423b438dc03e928a6dff86fd2cb/pyhive/tests/dbapi_test_case.py#L35
2. add a unit test for the _process_response function with and pass response

Reply:
OK, so you don't mind if I'm just testing for "not none" then - that's really what I was trying to establish here. If that's OK with you, I'll add that :-D

Reply:
Apart from the minor screwup with the missing ID on 7cda2cb (fixed by 575ea90), the tests look like they're working (apart from the ones that would be fixed by #312).

Is that the kind of thing you think you want?

Reply:
(also, I realised it was worth moving the id before anything that can fail, you want to capture it if you possibly can)